jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/api/client/clientrequestcontext/JAXRSClientIT.java

```diff
@@ -860,6 +861,53 @@ protected void checkFilterContext(ClientRequestContext context)
     assertContains(entity, "Accept");
   }
 
+  /*
+   * @testName: containsHeaderStringTest
+   * 
+   * @assertion_ids:  JAXRS:JAVADOC:1353; JAXRS:JAVADOC:1354; 
+   * 
+   * @test_Strategy: Check if the specified header contains a specified value.
+   * 
+   * ClientRequestFilter.abortWith
+   */
+  @Test
+  public void containsHeaderStringTest() throws Fault {
+    ContextProvider provider = new ContextProvider() {
+      @Override
+      protected void checkFilterContext(ClientRequestContext context) throws Fault {
+          assertTrue(context.containsHeaderString("header1", "value"::equalsIgnoreCase));
+          assertTrue(context.containsHeaderString("HEADER1", ",", "value2"::equals));
+          //Incorrect separator character
```

Comment:
IMHO a TCK test should only test for the existence of a given rule, not for the absence of particular bugs.

Reply:
I think this test is valid since it confirms that the value of the separator character parameter is actually used and taken into consideration by the impl.

Reply:
"Incorrect separator character" does not imply that. Checking for correct pickup of separator is implied in line 879 already.

Reply:
Like 879 only proves that the line separator parameter can be included in the invocation, but not that it is actually used by the impl.  The impl could be ignoring that parameter and accepting any separator character as valid.  This is why the "incorrect separator character" test is good.  It tests that the impl is using the the separator parameter and that it is checking it specifically and not just allowing any separator.

Reply:
I wonder if instead there could be a test that actively checks usage of the separator, and positively comments about that.

Reply:
I see your point, I'm just not sure there is a way to ensure that the separator parameter is actually being used without having at lease one test that expects failure in that respect.

Reply:
To me the test makes sense. If you split with the wrong separator (so you don't really split) then the assertion is false.

Reply:
That is correct, but just because using the wrong separator fails the test does not imply that using the correct separator passes the test.